components/clarinet-sdk/src-ts/bin/index.ts

```diff
@@ -0,0 +1,147 @@
+      child.on("exit", resolve);
+      child.on("error", reject);
+    });
+  }
+
+  console.log("\n");
+  console.log(
+    green("You are now ready to tests your smart contracts with Vitest and the Clarinet SDK")
+  );
+  console.log(green("Open ./tests/contract.test.ts to see an example"));
+}
+
+// check if Clarinet.toml exists and if the Node/NPM boilerplate doesn't
+async function checkIfProjectDirectoryIsValid() {
+  const isClarinetProject = fs.existsSync(path.join(process.cwd(), "Clarinet.toml"));
+  if (!isClarinetProject) {
+    throw new Error(
+      "Clarinet.toml not found in the current directory. Please run this command in a Clarinet project."
+    );
+  }
+
+  const unexpectedFiles = [
+    "package.json",
+    "vitest.config.js",
+    "vitest.config.ts",
+    "tsconfig.json",
+    "tests/contract.test.ts",
+  ];
+  for (const unexpectedFile of unexpectedFiles) {
+    if (fs.existsSync(path.join(process.cwd(), unexpectedFile))) {
+      const errorMsg = `A ${unexpectedFile} file already exists in this directory. It is possible that the testing framework has already been initialised.`;
+      throw new Error(errorMsg);
+    }
+  }
+
+  return true;
+}
+
+// copy the Node/NPM boilerplate
+function copyTemplateFiles(sdkBinDir: string) {
+  console.log("Copying package.json, tsconfig.json, vitest.config.js and sample test file");
+
+  fs.cpSync(path.join(sdkBinDir, "templates"), path.join(process.cwd(), "."), {
+    recursive: true,
+  });
+}
+
+// update to package.json name to  "<project-name>-tests"
+function updatePackageJSONProjectName(projectName: string) {
+  console.log("Updating package.json");
+  const packageJSONPath = path.join(process.cwd(), "package.json");
+  const packageJSON = JSON.parse(fs.readFileSync(packageJSONPath, "utf-8"));
+  packageJSON.name = `${projectName}-tests`;
+  fs.writeFileSync(packageJSONPath, JSON.stringify(packageJSON, null, 2));
+}
+
+// make sure we the current version of `@hirosystems/clarinet-sdk`
+function updateClarinetSDKVersion(sdkBinDir: string) {
+  const sdkPackageJSONPath = path.join(sdkBinDir, "../../../package.json");
+  const sdkPackageJSON = JSON.parse(fs.readFileSync(sdkPackageJSONPath, "utf-8"));
+  const version = sdkPackageJSON.version;
+
+  const projectPackageJSONPath = path.join(process.cwd(), "package.json");
+  const projectPackageJSON = JSON.parse(fs.readFileSync(projectPackageJSONPath, "utf-8"));
+  projectPackageJSON.dependencies["@hirosystems/clarinet-sdk"] = `^${version}`;
+
+  fs.writeFileSync(projectPackageJSONPath, JSON.stringify(projectPackageJSON, null, 2));
+}
+
+// add node and npm directories to the .gitignore
+function updateGitIgnore() {
+  console.log("Updating .gitignore");
+  const newLines = ["logs", "*.log", "npm-debug.log*", "coverage", "*.info", "node_modules"].join(
+    "\n"
+  );
+
+  const gitIgnorePath = path.join(process.cwd(), ".gitignore");
+  if (fs.existsSync(gitIgnorePath)) {
+    fs.appendFileSync(gitIgnorePath, `\n${newLines}`);
```

Comment:
Could add a comment to the resulting ignore file, so the user knows where these come from (why they are needed)

Reply:
Good call. Done L130 ✅